corehq/apps/app_manager/models.py

```diff
@@ -5707,63 +5707,89 @@ def reapply_overrides(self):
 
 
 def import_app(app_id_or_source, domain, source_properties=None, request=None, check_all_reports=True):
+    source_app = _get_source_app(app_id_or_source)
+    source = source_app.export_json(dump_json=False)
+
+    attachments = _get_attachments(source)
+    source['_attachments'] = {}
+
+    if source_properties is not None:
+        source.update(source_properties)
+
+    # Allow the wrapper to update to the current default build_spec
+    if 'build_spec' in source:
+        del source['build_spec']
+
+    app = _copy_app_from_source(domain, source)
```

Comment:
Realize you're working with existing code that uses the word "source", but I think `doc` is more explicit.

Reply:
I agree that the initially passed in `app_id_or_source` should be renamed to `app_id_or_doc`, but I'm having a hard time understanding what this `source` variable represents in this method because it doesn't seem like it is a doc. It really depends on what this line does `source = source_app.export_json(dump_json=False)` which doesn't give us back a doc right? 

I see `source` useful to describe the source/original/parent doc we are copying from, so it fits when used in `source_app` or `source_domain`. In this case it looks like it would be `source_json` but not sure if that is helpful to future readers.

Reply:
I'm not sure if we're splitting hairs over what exactly a doc is, but I'd say `export_json` returns a doc (at least when `dump_json` is false, otherwise it returns a string). It calls `to_json` on the app, which is essentially re-generating the doc, and then modifies it slightly.

Reply:
Interesting. So since the `app_id_or_doc` could be either, in the case where that is a doc, are we converting the doc to an `Application` and then back to a doc? Or should I think of the `Application` as a more specific doc? Hopefully that makes sense.

Reply:
> are we converting the doc to an Application and then back to a doc

Yes. Which sounds weird when said explicitly. The function accepts a source doc for the sake of the import app UI, where the user pastes in a doc. Then that doc is converted to a real app for the sake of being able to call `export_json`, which is needed so that `scrub_source` gets called and unique ids get replaced. Then it gets converted back to a doc because dicts are convenient to work with when you're copying a couch-based class, since you can directly manipulate all the attributes. I don't think it has to be this way - `scrub_source` could be a helper function instead of a method on `Application` - I think it's just how this code evolved.